### PR TITLE
[SPARK-42978][SQL] Derby&PG: RENAME cannot qualify a new-table-Name with a schema-Name

### PR DESCRIPTION
**Pull request by yaooqinn**
### What changes were proposed in this pull request?

Fix `rename a table` in derby and pg, which schema name is not allowed to qualify the new table name 

### Why are the changes needed?

bugfix

### Does this PR introduce _any_ user-facing change?

introduce a new error class

```json
  "CANNOT_RENAME_ACROSS_SCHEMA" : {
    "message" : [
      "Renaming a <type> across schemas is not allowed."
    ],
    "sqlState" : "0AKD0"
  },

```
### How was this patch tested?

new unit test

### CODE REVIEW

File: core/src/main/resources/error/error-classes.json [hunk 1]
```diff
@@ -129,6 +129,12 @@
     ],
     "sqlState" : "429BB"
   },
+  "CANNOT_RENAME_ACROSS_SCHEMA" : {
+    "message" : [
+      "Renaming a <type> across schemas is not allowed."
+    ],
+    "sqlState" : "0AKD0"
```
Comment (yaooqinn) [inline core/src/main/resources/error/error-classes.json:136]: this is borrowed from databricks runtime

**Comment by yaooqinn** [comment]
cc @cloud-fan @HyukjinKwon thanks

File: sql/core/src/main/scala/org/apache/spark/sql/jdbc/JdbcDialects.scala [hunk 2]
```diff
@@ -398,10 +398,24 @@ abstract class JdbcDialect extends Serializable with Logging {
    * @param newTable New name of the table.
    * @return The SQL statement to use for renaming the table.
    */
+  @deprecated("Please use renameTable method with identifiers", "3.5.0")
```
Comment (cloud-fan) [inline sql/core/src/main/scala/org/apache/spark/sql/jdbc/JdbcDialects.scala:401]:
```suggestion
  @deprecated("Please override renameTable method with identifiers", "3.5.0")
```

File: sql/core/src/main/scala/org/apache/spark/sql/jdbc/DB2Dialect.scala [hunk 3]
```diff
@@ -113,8 +114,9 @@ private object DB2Dialect extends JdbcDialect {
   // scalastyle:off line.size.limit
   // See https://www.ibm.com/support/knowledgecenter/en/SSEPGG_11.5.0/com.ibm.db2.luw.sql.ref.doc/doc/r0000980.html
   // scalastyle:on line.size.limit
-  override def renameTable(oldTable: String, newTable: String): String = {
-    s"RENAME TABLE $oldTable TO $newTable"
+  override def renameTable(oldTable: Identifier, newTable: Identifier): String = {
+    s" RENAME TABLE ${getFullyQualifiedQuotedTableName(oldTable)} TO " +
```
Comment (cloud-fan) [inline sql/core/src/main/scala/org/apache/spark/sql/jdbc/DB2Dialect.scala:118]: nit: why do we put a space at the beginning?

File: sql/core/src/main/scala/org/apache/spark/sql/jdbc/TeradataDialect.scala [hunk 4]
```diff
@@ -63,8 +64,9 @@ private case object TeradataDialect extends JdbcDialect {
   }
 
   // See https://docs.teradata.com/reader/scPHvjfglIlB8F70YliLAw/wysTNUMsP~0aGzksLCl1kg
-  override def renameTable(oldTable: String, newTable: String): String = {
-    s"RENAME TABLE $oldTable TO $newTable"
+  override def renameTable(oldTable: Identifier, newTable: Identifier): String = {
+    s" RENAME TABLE ${getFullyQualifiedQuotedTableName(oldTable)} TO " +
```
Comment (cloud-fan) [inline sql/core/src/main/scala/org/apache/spark/sql/jdbc/TeradataDialect.scala:68]: ditto

**Review by cloud-fan — APPROVED** [review]

**Pull request closed without merging**